File: demo/src/main/java/com/example/demo/Service/CompanyService.java
package com.example.demo.Service;

import java.util.List;
import java.util.Optional;

import com.example.demo.Entity.Company;

public interface CompanyService {
    Company addCompany(Company company, Long userId);
    Company updateCompany(Long id, Company company);
    void deleteCompany(Long id);
    List<Company> getAllCompanies();
    List<Company> getCompaniesByStatus(Boolean isActive);
    List<Company> getCompaniesByUserId(Long userId);
    Optional<Company> getCompanyById(Long id);

    // 🔹 Add this method for AdminController
    Company toggleCompanyStatus(Long id, boolean status);
}

File: company-frontend/src/app/service/company.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Company } from '../model/company';

@Injectable({
  providedIn: 'root'
})
export class CompanyService {

  private baseUrl = 'http://localhost:8082/api/companies'; // adjust if needed

  constructor(private http: HttpClient) {}

  // Add a new company
  addCompany(company: Company): Observable<Company> {
    return this.http.post<Company>(this.baseUrl, company);
  }

  // Update company by ID
  updateCompany(id: number, company: Company): Observable<Company> {
    return this.http.put<Company>(`${this.baseUrl}/${id}`, company);
  }

  // Delete company by ID
  deleteCompany(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }

  // Get all companies
  getAllCompanies(): Observable<Company[]> {
    return this.http.get<Company[]>(this.baseUrl);
  }

  // Get companies by status
  getCompaniesByStatus(isActive: boolean): Observable<Company[]> {
    return this.http.get<Company[]>(`${this.baseUrl}/status/${isActive}`);
  }

  // Get all company details
  getAllCompanyDetails(): Observable<Company[]> {
    return this.http.get<Company[]>(`${this.baseUrl}/details`);
  }
}
